src/Compiler/Checking/import.fs

```diff
@@ -66,556 +67,710 @@ type ImportMap(g: TcGlobals, assemblyLoader: AssemblyLoader) =
+            error (
+                Error(
+                    FSComp.SR.impReferencedTypeCouldNotBeFoundInAssembly (
+                        String.concat "." (Array.append path [| typeName |]),
+                        ccu.AssemblyName
+                    ),
+                    m
+                )
+            )
 #if !NO_TYPEPROVIDERS
     // Validate (once because of caching)
     match tycon.TypeReprInfo with
     | TProvidedTypeRepr info ->
-            //printfn "ImportTypeRefData: validating type: typeLogicalName = %A" typeName
-            ValidateProvidedTypeAfterStaticInstantiation(m, info.ProvidedType, path, typeName)
-    | _ -> 
-            ()
+        //printfn "ImportTypeRefData: validating type: typeLogicalName = %A" typeName
+        ValidateProvidedTypeAfterStaticInstantiation(m, info.ProvidedType, path, typeName)
+    | _ -> ()
 #endif
-    match tryRescopeEntity ccu tycon with 
-    | ValueNone -> error (Error(FSComp.SR.impImportedAssemblyUsesNotPublicType(String.concat "." (Array.toList path@[typeName])), m))
+    match tryRescopeEntity ccu tycon with
+    | ValueNone -> error (Error(FSComp.SR.impImportedAssemblyUsesNotPublicType (String.concat "." (Array.toList path @ [ typeName ])), m))
     | ValueSome tcref -> tcref
-    
 
 /// Import a reference to a type definition, given an AbstractIL ILTypeRef, without caching
 //
 // Note, the type names that flow to the point include the "mangled" type names used for static parameters for provided types.
 // For example,
 //       Foo.Bar,"1.0"
-// This is because ImportProvidedType goes via Abstract IL type references. 
-let ImportILTypeRefUncached (env: ImportMap) m (tref: ILTypeRef) = 
-    let path, typeName = 
-        match tref.Enclosing with 
-        | [] -> 
-            splitILTypeNameWithPossibleStaticArguments tref.Name
-        | h :: t -> 
+// This is because ImportProvidedType goes via Abstract IL type references.
+let ImportILTypeRefUncached (env: ImportMap) m (tref: ILTypeRef) =
+    let path, typeName =
+        match tref.Enclosing with
+        | [] -> splitILTypeNameWithPossibleStaticArguments tref.Name
+        | h :: t ->
             let nsp, tname = splitILTypeNameWithPossibleStaticArguments h
             // Note, subsequent type names do not need to be split, only the first
             [| yield! nsp; yield tname; yield! t |], tref.Name
 
     ImportTypeRefData (env: ImportMap) m (tref.Scope, path, typeName)
 
-    
 /// Import a reference to a type definition, given an AbstractIL ILTypeRef, with caching
 let ImportILTypeRef (env: ImportMap) m (tref: ILTypeRef) =
     match env.ILTypeRefToTyconRefCache.TryGetValue tref with
     | true, tcref -> tcref
     | _ ->
-        let tcref = ImportILTypeRefUncached  env m tref
-        env.ILTypeRefToTyconRefCache[tref] <- tcref
+        let tcref = ImportILTypeRefUncached env m tref
+        env.ILTypeRefToTyconRefCache[ tref ] <- tcref
```

Comment:
@nojaf Is this change expected?

Reply:
I don't think so no, please open an issue for this.
Ref: https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-indexslice-expressions